rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -2389,24 +2543,28 @@ error_code sceNpManagerGetOnlineName(vm::ptr<SceNpOnlineName> onlineName)
 		return SCE_NP_ERROR_INVALID_ARGUMENT;
 	}
 
-	if (g_psn_connection_status == SCE_NP_MANAGER_STATUS_OFFLINE)
+	if (nph->get_psn_status() == SCE_NP_MANAGER_STATUS_OFFLINE)
 	{
 		return not_an_error(SCE_NP_ERROR_OFFLINE);
 	}
 
-	if (g_psn_connection_status != SCE_NP_MANAGER_STATUS_ONLINE)
+	if (nph->get_psn_status() != SCE_NP_MANAGER_STATUS_ONLINE)
 	{
 		return SCE_NP_ERROR_INVALID_STATE;
 	}
 
+	memcpy(onlineName.get_ptr(), &nph->get_online_name(), sizeof(SceNpOnlineName));
```

Comment:
Nitpick, but `sizeof(nph->get_online_name())` may work. Or something similar, to tie the size to this function's return type.

Reply:
onlineName.size() works best.